src/common-rest-api/src/main/java/org/niis/xroad/restapi/controller/SessionStatusController.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * The MIT License
+ *
+ * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.controller;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.niis.xroad.restapi.openapi.ControllerUtil;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+
+@RestController
+@RequestMapping(SessionStatusController.NOTIFICATIONS_API_V1_PATH + "/session-status")
+@Slf4j
+@PreAuthorize("isAuthenticated()")
+@RequiredArgsConstructor
+public class SessionStatusController {
+
+
+    public static final String NOTIFICATIONS_API_V1_PATH = ControllerUtil.API_V1_PREFIX + "/notifications";
```

Comment:
Feels a bit sketchy that this part of the `/notifications` api is in this controller (in the common api), but some parts of the api remain only in `proxy-ui-api`. I wonder if this would be enough reason to move the session polling in to its own endpoint instead of `/notifications`. 

What do you think? Moving the session polling to another path is a breaking change but then again it is supposed to be used by the frontend only (and not by some arbitrary api-client).

Reply:
The NotificationApiDisableSessionTimeoutFilter enables making requests without pushing the session timeout further. That filter is based on certain API path - `/notifications`.  
That feature is needed for these periodic status polling requests. If endpoints are not concentrated to a single path, this filter has to be adjusted accordingly.  This kind of filter is a awkwardly hidden dependency. i.e. It's not apparent when you look in the controller that there is special session timeout handling applied for these requests. 

We could naturally make the filter select the polling requests according to some other rule. e.g. special HTTP header or some URL parameter values. But it poses new requirements then to the endpoint user -

Reply:
What about changing the path of the endpoint  from `/notifications` to  perhaps more descriptive `/polling` ?

Reply:
Or maybe the `NotificationsApiController` class under proxy-ui-api could be renamed `NotificationsAlertsApiController` since it only handles `/notifications/alerts` now. And the `SessionStatusController` under common-api could be `NotificationsSessionStatusApiController` since it only handles `/notifications/session-status` path. Perhaps that would add more relation between these controllers (?)

Reply:
Done in : 7179dad  & 
4d6b2da